### PR TITLE
Adds plotter utility

### PR DESCRIPTION
**Pull request by premalrupnur**
Followup of #23. Completes #3 
Am I heading right

### CODE REVIEW

**Comment by premalrupnur** [comment]
Once I complete the correct implementation of the function I would add  example for this too

File: examples/plotter.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+import torch.nn.functional as F
+from torch.cuda import amp
+import pytest
+from tqdm import tqdm
+import torchvision.transforms as T
+from pytorch_cnn_trainer import dataset
+from pytorch_cnn_trainer import model_factory
+from pytorch_cnn_trainer import utils
+from pytorch_cnn_trainer import engine
+from torch.optim.swa_utils import SWALR
```
Comment (oke-aditya) [inline examples/plotter.py]: May I know why so many imports are needed?

Can this file be in the library instead fo `examples` into `pytorch_cnn_trainer`?

File: examples/plotter.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+import matplotlib.pyplot as plt 
+import config
+
+device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+
+def plot(history,metric1,metric2):
```
Comment (oke-aditya) [inline examples/plotter.py]: Can you attach some O/P of these graphs. it will make it understand what it does.

File: pytorch_cnn_trainer/engine.py [hunk 3]
```diff
@@ -394,8 +394,9 @@ def fit(
 
     history = {
         "train": {"top1_acc": top1_acc, "top5_acc": top5_acc, "loss": loss},
-        "val": {"top1_acc": loss_v, "top5_acc": top5_acc_v, "loss": top1_acc_v},
+        "val": {"top1_acc":top1_acc_v, "top5_acc": top5_acc_v, "loss":loss_v }
```
Comment (oke-aditya) [inline pytorch_cnn_trainer/engine.py:397]: Good correction

**Review by oke-aditya — CHANGES_REQUESTED** [review]

File: examples/plotter.py (hunk 2 above)
Comment (oke-aditya) [reply examples/plotter.py]: I think function prototype is fine, It should do the job.

File: examples/plotter.py (hunk 1 above)
Comment (premalrupnur) [reply examples/plotter.py]: Actually I too was going to remove those imports but then forgot about it , , ,will do

File: examples/plotter.py (hunk 2 above)
Comment (premalrupnur) [reply examples/plotter.py]: Ok will do

Comment (premalrupnur) [reply examples/plotter.py]: > Can you attach some O/P of these graphs. it will make it understand what it does.

Screenshot (209)

Comment (premalrupnur) [reply examples/plotter.py]: tried just on two epochs so a linear one

File: pytorch_cnn_trainer/plotter.py [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import matplotlib.pyplot as plt 
+
```
Comment (oke-aditya) [inline pytorch_cnn_trainer/plotter.py:2]: Can we cover this up with `__all__` ?

File: pytorch_cnn_trainer/plotter.py [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+def plot(history,metric1,metric2):
```
Comment (oke-aditya) [inline pytorch_cnn_trainer/plotter.py]: We should probably give a better name 🙂

**Review by oke-aditya — CHANGES_REQUESTED** [review]

File: pytorch_cnn_trainer/plotter.py (hunk 4 above)
Comment (premalrupnur) [reply pytorch_cnn_trainer/plotter.py:2]: yes I'll do it but do i have to add a line for this file in __init__.py
I mean This line - `from pytorch_cnn_trainer.plotter import *`

Comment (oke-aditya) [reply pytorch_cnn_trainer/plotter.py:2]: Yes you should modify `init` file too.

Also include this in tests 👍 

Test can be simply to check if this function is called. I would rather prefer to return a plotter object that user can re-use to get the plot

**Comment by premalrupnur** [comment]
Is there a need to add a chart to get the summary of entire train_metrics if it can be analyzed by current function
like user would be able to analyze the training by the two parameters we have passed in current function

File: pytorch_cnn_trainer/plotter.py (hunk 5 above)
Comment (premalrupnur) [reply pytorch_cnn_trainer/plotter.py]: I probably don't know any better than this 🙂,any suggestions by your side

File: pytorch_cnn_trainer/plotter.py [hunk 6]
```diff
@@ -1,12 +1,15 @@
...
+    train_m1 = plt.plot(history["train"][metric1])
+    train_m2 = plt.plot(history["train"][metric2])
+    val_m1 = plt.plot(history["val"][metric1])
+    val_m2 = plt.plot(history["val"][metric2])
     plt.title('train v/s val')
     plt.ylabel('metrics')
     plt.xlabel('epoch')
     plt.legend(['train-'+ metric1,'train-'+metric2, 'val-'+metric1,'val-'+metric2], loc='upper left')
     plt.show()
+    return [train_m1,train_m2,val_m1,val_m2]
```
Comment (premalrupnur) [inline pytorch_cnn_trainer/plotter.py]: I think I have returned the object incorrectly

Comment (oke-aditya) [reply pytorch_cnn_trainer/plotter.py]: Yes. Let's return None to this ;) 

Write these down in tests, rest looks fine.

Comment (premalrupnur) [reply pytorch_cnn_trainer/plotter.py]: Can you just give a hint what did you actually expect as a returned object and I hope it does'nt seem hardcoded now

Comment (premalrupnur) [reply pytorch_cnn_trainer/plotter.py]: > Write these down in tests, rest looks fine.

I have written test for it

**Comment by premalrupnur** [comment]
Hey how train_sanity_fit would return metrics because it is not evaluating anything other than loss so only loss can be returned ig
I would like to work on it today

File: pytorch_cnn_trainer/plotter.py (hunk 5 above)
Comment (oke-aditya) [reply pytorch_cnn_trainer/plotter.py]: Can your rename it from plot to `plot_results`
It will clash with `plt.plot`

**Comment by oke-aditya** [comment]
Let's dicuss sanity in new PR.
I just need 1 rename.

**Comment by premalrupnur** [comment]
> I just need 1 rename.

1 rename for what , 'plot' function

**Comment by premalrupnur** [comment]
> Let's dicuss sanity in new PR.
ok but I wish to complete my work on sanity_fit pr today itself cause next week I have exams so have to dedicate the time for it and by that time hactoberfest would also get over ,so , I hope you understand

**Comment by oke-aditya** [comment]
Yes change it to `plot_results`

**Review by oke-aditya — APPROVED** [review]

**Comment by premalrupnur** [comment]
Does this pr meet all the requirements for it to be merged??

**Comment by oke-aditya** [comment]
Yes, but I need to have a look once.

**Comment by premalrupnur** [comment]
> Let's dicuss sanity in new PR.

ok,but can you just give a me slight idea I would just make changes and create pr once this one is closed
